pkg/sds/server/reconciler.go

```diff
@@ -143,15 +149,24 @@ func (d *DataplaneReconciler) generateSnapshot(dataplane *mesh_core.DataplaneRes
 	snap := envoy_cache.Snapshot{
 		Resources: [envoy_types.UnknownType]envoy_cache.Resources{},
 	}
-	snap.Resources[envoy_types.Secret] = envoy_cache.NewResources(version, []envoy_types.Resource{
-		identitySecret.ToResource(IdentityCertResource),
-		caSecret.ToResource(MeshCaResource),
-	})
+	secrets = append(secrets, caSecret.ToResource(envoy_names.MeshCaResource))
+	snap.Resources[envoy_types.Secret] = envoy_cache.NewResources(version, secrets)
 	return snap, nil
 }
 
 func (d *DataplaneReconciler) updateInsights(dataplaneId core_model.ResourceKey, snapshot envoy_cache.Snapshot) error {
-	secret := snapshot.Resources[envoy_types.Secret].Items[IdentityCertResource].(*envoy_auth.Secret)
+	var secret *envoy_auth.Secret
+	dpCerts := 0
+	for _, resource := range snapshot.Resources[envoy_types.Secret].Items {
+		sec := resource.(*envoy_auth.Secret)
+		if envoy_names.IsDpCertResource(sec.Name) {
+			secret = sec
```

Comment:
Is it correct that `dataplaneInsight` will have expiration time only of the dp cert of the last outbound? Is it okay?

Reply:
all certs are generated at the same time, so I'd say - yes